### PR TITLE
shared/ConnId: Fix logic error in constructor

### PR DESCRIPTION
**Pull request by chrysn**
This fixes a copy-paste error stemming from using code from from_int_raw without inverting the conditions, and adds documentation including doctests on this and related methods.

---

This fixes a bug introduced in #260 that was not covered by tests (because the tests do not try the path where the initiator deliberately chooses a C_I)

### CODE REVIEW

File: shared/src/lib.rs [hunk 1]
```diff
@@ -132,6 +132,23 @@ impl ConnId {
     }
 
     /// The CBOR encoding of the identifier.
+    ///
+    /// For short the 48 compact connection identifiers, this is identical to the slice
```
Comment (geonnave) [inline shared/src/lib.rs:136]: > For short the 48 compact connection identifiers

I don't understand this part.

Comment (geonnave) [reply shared/src/lib.rs:136]: Maybe just replace it by `For compact connection identifiers (integer value in -24..=23)`. Referring to the range is easier to track to the RFC if needed.

**Comment by geonnave** [comment]
Looks good, left one comment, thanks.

File: shared/src/lib.rs (hunk 1 above)
Comment (chrysn) [reply shared/src/lib.rs]: Good point, fix forcepushed.

**Pull request merged**